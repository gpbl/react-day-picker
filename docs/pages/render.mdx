import { Render } from '@components/Render';
import { getSrc } from '@utils/getSrc';
import * as examples from '@examples/index';

export const getStaticProps = () =>
  getSrc([
    'ContainerAttributes.tsx',
    'Controlled.tsx',
    'CssVariables.tsx',
    'CustomCaption.tsx',
    'CustomDay.tsx',
    'CustomDisableRow.tsx',
    'CustomMultiple.tsx',
    'CustomSingle.tsx',
    'Popover.tsx',
    'DefaultMonth.tsx',
    'Disabled.tsx',
    'DisableNavigation.tsx',
    'Dropdown.tsx',
    'DropdownButtons.tsx',
    'DropdownMultipleMonths.tsx',
    'Fixedweeks.tsx',
    // 'FocusRecursive.tsx',
    // 'Formatters.tsx',
    // 'FromToMonth.tsx',
    // 'FromToYear.tsx',
    // 'InputRange.tsx',
    // 'InputTime.tsx',
    // 'Keyboard.tsx',
    // 'ModifiersClassnames.tsx',
    // 'ModifiersCustom.tsx',
    // 'ModifiersDisabled.tsx',
    // 'ModifiersHidden.tsx',
    // 'ModifiersSelected.tsx',
    // 'ModifiersStyle.tsx',
    // 'ModifiersToday.tsx',
    // 'Multiple.tsx',
    // 'MultipleMinMax.tsx',
    // 'MultipleMonths.tsx',
    // 'MultipleMonthsId.tsx',
    // 'MultipleMonthsPaged.tsx',
    // 'NumberingSystem.tsx',
    // 'OutsideDays.tsx',
    'Range.tsx',
    'RangeMinMax.tsx',
    'RangeShiftKey.tsx'
    // 'Rtl.tsx',
    // 'Single.tsx',
    // 'SingleRequired.tsx',
    // 'Spanish.tsx',
    // 'SpanishWeekStartsOn.tsx',
    // 'Start.tsx',
    // 'StylingCss.tsx',
    // 'StylingCssModules.tsx',
    // 'StylingInline.tsx',
    // 'StylingModifiers.tsx',
    // 'Testcase1567.tsx',
    // 'WeekIso.tsx',
    // 'Weeknumber.tsx',
    // 'WeeknumberCustom.tsx',
    // 'WeeknumberIso.tsx'
  ]);

<Render />
