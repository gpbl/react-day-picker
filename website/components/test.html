<!DOCTYPE html>
<html>
<head>
<style>
body {
  display: flex;
  margin: 0;
  padding: 0;
  min-height: 100vh;
  overflow-x: hidden;
}

.sidebar {
  position: sticky;
  top: 0;
  height: 100vh;
  width: 200px;
  background-color: #f5f5f5;
  padding: 20px;
  overflow-y: auto;
}

.main-content {
  flex-grow: 1;
  padding: 20px;
}
</style>
</head>
<body>

<div class="sidebar">
  <h2>Sidebar</h2>
  <p>This is a sticky sidebar.</p>
</div>

<div class="main-content">
  <h2>Main Content</h2>
  <p>Scroll the page.</p>
  <!-- Add enough content here so that the page scrolls -->
</div>

</body>
</html>
