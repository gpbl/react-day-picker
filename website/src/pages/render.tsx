/* eslint-disable @typescript-eslint/no-var-requires */
import React from 'react';
import { useLocation } from '@docusaurus/router';
import BrowserOnly from '@docusaurus/BrowserOnly';

/**
 * Create a page at `/render` URL that renders the examples from the examples
 * `directory`.
 * */
export default function Render(): JSX.Element {
  const location = useLocation();
  return (
    <BrowserOnly>
      {() => {
        const fileName = new URLSearchParams(location.search).get('example');
        try {
          require(`../../examples/${fileName}`).default;
        } catch (e) {
          console.error('Error requiring %s', `../../examples/${fileName}`, e);
          return <pre>{e.message}</pre>;
        }
        const Component = require(`../../examples/${fileName}`).default;
        return <Component />;
      }}
    </BrowserOnly>
  );
}
