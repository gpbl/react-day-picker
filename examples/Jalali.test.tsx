import React from "react";

import { render, screen } from "@/test/render";

import { Jalali } from "./Jalali";

const today = new Date(2021, 10, 25);

beforeAll(() => {
  jest.setSystemTime(today);
});

afterAll(() => {
  jest.useRealTimers();
});

beforeEach(() => {
  render(
    <div role="application">
      <Jalali />
    </div>
  );
});

test("should match the snapshot", () => {
  expect(screen.getByRole("application")).toMatchSnapshot();
});
