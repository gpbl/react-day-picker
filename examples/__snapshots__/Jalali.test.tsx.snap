// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div
  role="application"
>
  <div
    class="rdp-calendar"
    dir="rtl"
    id=":r1:"
  >
    <div
      class="rdp-months"
    >
      <div
        class="rdp-nav"
        role="toolbar"
      >
        <button
          aria-controls=":r1:"
          aria-label="Previous Month"
          class="rdp-button_previous"
          name="previous-month"
          type="button"
        >
          <svg
            class="rdp-chevron"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <polygon
              points="15 17.23 9.43 11.5 15 5.76 13.28 4 6 11.5 13.28 19"
            />
          </svg>
        </button>
        <button
          aria-controls=":r1:"
          aria-label="Next Month"
          class="rdp-button_next"
          name="next-month"
          type="button"
        >
          <svg
            class="rdp-chevron"
            height="24"
            viewBox="0 0 24 24"
            width="24"
          >
            <polygon
              points="8 18.23 13.57 12.5 8 6.76 9.72 5 17 12.5 9.72 20"
            />
          </svg>
        </button>
      </div>
      <div
        class="rdp-month_wrapper"
      >
        <div
          class="rdp-month_caption"
          id=":r1:-caption-0"
        >
          <span
            aria-label=""
            aria-live="polite"
            class="rdp-caption_label"
            role="status"
          >
            نوامبر 2021
          </span>
        </div>
        <div
          aria-labelledby=":r1:-caption-0"
          aria-multiselectable="false"
          class="rdp-month"
          id=":r1:-grid-0"
          role="grid"
        >
          <div
            aria-rowindex="1"
            class="rdp-weekdays"
            role="row"
          >
            <span
              aria-colindex="1"
              aria-label="شنبه"
              class="rdp-weekday"
              role="columnheader"
            >
              ش
            </span>
            <span
              aria-colindex="2"
              aria-label="یکشنبه"
              class="rdp-weekday"
              role="columnheader"
            >
              1ش
            </span>
            <span
              aria-colindex="3"
              aria-label="دوشنبه"
              class="rdp-weekday"
              role="columnheader"
            >
              2ش
            </span>
            <span
              aria-colindex="4"
              aria-label="سه‌شنبه"
              class="rdp-weekday"
              role="columnheader"
            >
              3ش
            </span>
            <span
              aria-colindex="5"
              aria-label="چهارشنبه"
              class="rdp-weekday"
              role="columnheader"
            >
              4ش
            </span>
            <span
              aria-colindex="6"
              aria-label="پنجشنبه"
              class="rdp-weekday"
              role="columnheader"
            >
              5ش
            </span>
            <span
              aria-colindex="7"
              aria-label="جمعه"
              class="rdp-weekday"
              role="columnheader"
            >
              ج
            </span>
          </div>
          <div
            class="rdp-weeks"
            role="rowgroup"
          >
            <div
              aria-rowindex="2"
              class="rdp-week"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-hidden="true"
                aria-label=""
                class="rdp-day rdp-hidden rdp-outside"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  30
                </span>
              </div>
              <div
                aria-colindex="2"
                aria-hidden="true"
                aria-label=""
                class="rdp-day rdp-hidden rdp-outside"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  31
                </span>
              </div>
              <div
                aria-colindex="3"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  1
                </span>
              </div>
              <div
                aria-colindex="4"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  2
                </span>
              </div>
              <div
                aria-colindex="5"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  3
                </span>
              </div>
              <div
                aria-colindex="6"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  4
                </span>
              </div>
              <div
                aria-colindex="7"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  5
                </span>
              </div>
            </div>
            <div
              aria-rowindex="3"
              class="rdp-week"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  6
                </span>
              </div>
              <div
                aria-colindex="2"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  7
                </span>
              </div>
              <div
                aria-colindex="3"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  8
                </span>
              </div>
              <div
                aria-colindex="4"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  9
                </span>
              </div>
              <div
                aria-colindex="5"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  10
                </span>
              </div>
              <div
                aria-colindex="6"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  11
                </span>
              </div>
              <div
                aria-colindex="7"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  12
                </span>
              </div>
            </div>
            <div
              aria-rowindex="4"
              class="rdp-week"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  13
                </span>
              </div>
              <div
                aria-colindex="2"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  14
                </span>
              </div>
              <div
                aria-colindex="3"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  15
                </span>
              </div>
              <div
                aria-colindex="4"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  16
                </span>
              </div>
              <div
                aria-colindex="5"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  17
                </span>
              </div>
              <div
                aria-colindex="6"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  18
                </span>
              </div>
              <div
                aria-colindex="7"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  19
                </span>
              </div>
            </div>
            <div
              aria-rowindex="5"
              class="rdp-week"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  20
                </span>
              </div>
              <div
                aria-colindex="2"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  21
                </span>
              </div>
              <div
                aria-colindex="3"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  22
                </span>
              </div>
              <div
                aria-colindex="4"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  23
                </span>
              </div>
              <div
                aria-colindex="5"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  24
                </span>
              </div>
              <div
                aria-colindex="6"
                aria-label=""
                class="rdp-day rdp-focusable rdp-today"
                role="gridcell"
                tabindex="0"
              >
                <span
                  class="rdp-day_date"
                >
                  25
                </span>
              </div>
              <div
                aria-colindex="7"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  26
                </span>
              </div>
            </div>
            <div
              aria-rowindex="6"
              class="rdp-week"
              role="row"
            >
              <div
                aria-colindex="1"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  27
                </span>
              </div>
              <div
                aria-colindex="2"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  28
                </span>
              </div>
              <div
                aria-colindex="3"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  29
                </span>
              </div>
              <div
                aria-colindex="4"
                aria-label=""
                class="rdp-day rdp-focusable"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  30
                </span>
              </div>
              <div
                aria-colindex="5"
                aria-hidden="true"
                aria-label=""
                class="rdp-day rdp-hidden rdp-outside"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  1
                </span>
              </div>
              <div
                aria-colindex="6"
                aria-hidden="true"
                aria-label=""
                class="rdp-day rdp-hidden rdp-outside"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  2
                </span>
              </div>
              <div
                aria-colindex="7"
                aria-hidden="true"
                aria-label=""
                class="rdp-day rdp-hidden rdp-outside"
                role="gridcell"
                tabindex="-1"
              >
                <span
                  class="rdp-day_date"
                >
                  3
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
